independent-projects/resteasy-reactive/server/vertx-java-io/src/main/java/io/quarkus/vertx/java/io/VertxBufferImpl.java

```diff
@@ -0,0 +1,490 @@
+package io.quarkus.vertx.java.io;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.util.CharsetUtil;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.impl.Arguments;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+public class VertxBufferImpl implements Buffer {
```

Comment:
Why is this needed? Don't we already have this class elsewhere?

Reply:
Thanks for the hint. This one was supposed to be moved from `independent-projects/resteasy-reactive/server/vertx/src/main/java/org/jboss/resteasy/reactive/server/vertx/VertxBufferImpl.java`, but I forgot to remove the old one. Which I am going to do right now.

There is still one more in `extensions/vertx/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxBufferImpl.java` which can be removed in favor of this one. Which I am going to do as well.

Reply:
I would like to ask you to add a little Javadoc to the classes you are going to use in CXF, as these classes are now no longer going to be internal, so it would be nice for integrators to know a little about what each one does.

Reply:
Hm... now I see there is also `io.vertx.core.buffer.impl.BufferImpl` that does some additional bounds checks. I guess the two `VertxBufferImpl` do not do it intentionally for performance reasons?

Reply:
> I would like to ask you to add a little Javadoc 

Haha, I can try

Reply:
🙏🏼

Reply:
> I guess the two VertxBufferImpl do not do it intentionally for performance reasons?

OK, if so, I guess, it won't be so straightforward to move the refactored classes to Vert.x due to our "unsafe" `BufferImpl`. Therefore, I am going to rename the module to `quarkus-vertx-java-io` in line with the expectation that it will stay in Quarkus. Also, because the module is now used also in `quarkus-vertx` and `quarkus-websockets-next`, I am going to move it to directly under `independent-projects`.